### ISSUE TITLE
jbad buildings do not work well with ALiVE

### ISSUE THREAD

**Issue by NZF-JDWang** [M1lkm8n/jbad#27]
As mentioned in the BI Forums it seems like there is an issue with maps that use jbad buildings and the ALiVE system.

The best example is the Diyala map which used to use jbad and then switch to the CUP version. The entire time it used jbad no matter how many times the map was indexed for ALiVE you could not get units to spawn properly. The ALiVE system would show that the map was indexed correctly however you would only see 1 or 2 units spawning.

When the map was changed to use the CUP versions of the buildings it was reindexed again and instantly ALiVE could spawn units all over the map.

Lythium is currently suffering from the same issue. You can get some units to spawn with ALiVE but by and large the map is full of dead spots and it renders the map almost useless for groups using ALiVE.

I've only seen this issue with maps that use jbad buildings, so I assume there must be something different about them. which stops ALiVE from registering them?

**Comment by M1lkm8n** [comment]
i think possibly its our naming scheme. Talk to someone from the ALIVE team but i am betting that is the issue. CUP uses the original names. Jbad does not. We do however have all the AI spawn points in a majority of the buildings

**Comment by HeroesandvillainsOS** [comment]
M1lkm8n thanks for the reply. I’ve forwarded this over to our Dev Slack so hopefully when one of our lead devs is available they can offer an opinion on the matter.

**Comment by tupolov** [comment]
Its not naming. Units spawning in and around buildings relies on two things:

1. Correct indexing of buildings for the map
2. Correct house positions configured

Point 2 is important, previously if buildings did not return house positions units would not spawn.

I suspect the index was incorrect?

**Comment by NZF-JDWang** [comment]
I find it hard to believe that it's just the index, I'm fairly sure it'd been indexed more than once. Especially when diyala was indexed multiple times and had the exact same issue right up till they switched from jbad to the cup versions of the buildings.

Unless the cup guys did something other than changing the names?

**Comment by marceldev89** [comment]
Diyala did get a new index by @HeroesandvillainsOS just after they released the update that switched to CUP. There also were some fixes to the indexing code between the first and last Diyala index which might be related.

As for Lythium, I guess it could be worth a try to do a fresh index but I'm not volunteering. :grin:

**Comment by Lifetap** [comment]
I've also experienced issues with ALIVE civilians on Lythium. 
Firstly the number of civilians spawning is very small, despite trying the "extreme" setting and a very limited zone (1km x 1km).  I was rarely able to achieve more than 1-3 civilians.
Secondly, civilians would spawn in an area, and then most (but not all) would very rapidly despawn (within 30 seconds) .  No matter how I moved the player and/or zeus camera I could never get the original despawned civilians to reappear.  Even when leaving the area (where all civilians would despawn) and then returning, the original despawned civilians would not reappear.  It was as though their position was deemed inappropriate, and they were permanently removed.
I was using a very stripped down mission, which was then tested on Altis, and worked perfectly.
